cmd/csaf_checker/roliecheck.go

```diff
@@ -52,35 +73,87 @@ func tlpLabel(label *csaf.TLPLabel) csaf.TLPLabel {
 	return csaf.TLPLabelUnlabeled
 }
 
+// extractTLP extracts the tlp label of the given document
+// and defaults to TLP:WHITE if not found.
+func (p *processor) extractTLP(doc any) csaf.TLPLabel {
+	advisoryTLP, err := p.expr.Eval(`$.document.distribution.tlp.label`, doc)
+	if err != nil {
+		return csaf.TLPLabelUnlabeled
+	}
+	label, ok := advisoryTLP.(string)
+	if !ok {
+		return csaf.TLPLabelUnlabeled
+	}
+	return csaf.TLPLabel(label)
+}
+
+// evaluateTLP extracts the TLP label from a given document and
+// calls upon functions further checking for mistakes in access-protection
+// or label assignment between feed and advisory
+func (lc *labelChecker) evaluateTLP(p *processor, doc any, url string) {
+
+	advisoryLabel := p.extractTLP(doc)
+
+	// If the client has no authorization it shouldn't be able
+	// to access TLP:AMBER or TLP:RED advisories
+	if !p.usedAuthorizedClient() &&
+		(advisoryLabel == csaf.TLPLabelAmber || advisoryLabel == csaf.TLPLabelRed) {
+		p.badAmberRedPermissions.use()
+		p.badAmberRedPermissions.error(
+			"Advisory %s of TLP level %v is not access protected.",
+			url, advisoryLabel)
+	}
```

Comment:
This error message and the one from line 171/174 get both triggered, when a TLP:RED document is in a TLP:WHITE feed. I would expect only one message...

Reply:
Yes, this should not happen. I guess the hole case block of checking `label == csaf.TLPLabelWhite` in `checkProtection` is not correct any more. Removing it with the ` p.badWhitePermissions.use()` to be
moved to `evaluateTLP` line 100 should be the solution ->  @JanHoefelmeyer

Reply:
I agree,  addressed this in https://github.com/csaf-poc/csaf_distribution/pull/391/commits/89e7439145a658dcc3d2c957d6eb90f63097178e

Reply:
@tschmidtb51 This should be better now. Please re-test.

Reply:
@s-l-teichmann Unfortunately, I'm currently not near my test environment - I'll look into it tonight.